halo2_proofs/src/plonk/prover.rs

```diff
@@ -353,16 +357,26 @@ where
                 );
 
                 // Add blinding factors to advice columns
-                for advice_values in &mut advice_values {
+                for (column_index, advice_values) in &mut advice_values.iter_mut().enumerate() {
                     for cell in &mut advice_values[unusable_rows_start..] {
                         *cell = Scheme::Scalar::random(&mut rng);
+                        if witness.unblinded_advice.contains(&column_index) {
+                            *cell = Blind::default().0;
+                        } else {
+                            *cell = Scheme::Scalar::random(&mut rng);
+                        }
```

Comment:
I think we should use `column_indices` instead of `enumerate()` considering multi-phase, also there is a `random` leftover. And I'd prefer to just keep unblinded cells as `0` and add a sanity check they are indeed `0`.

```suggestion
                for (column_index, advice_values) in column_indices.iter().zip(&mut advice_values) {
                    if witness.unblinded_advice.contains(column_index) {
                        for cell in &mut advice_values[unusable_rows_start..] {
                            *cell = Scheme::Scalar::random(&mut rng);
                        }
                    } else {
                        #[cfg(feature = "sanity-checks")]
                        for cell in &advice_values[unusable_rows_start..] {
                            assert_eq!(*cell, Scheme::Scalar::ZERO);
                        }
                    }
                }
```

Reply:
Queried columns are not sorted by phase, so we need to use the correct index (or id) of them instead using their index in each phase